XenAdmin/MainWindow.cs

```diff
@@ -979,6 +979,34 @@ void connection_CachePopulated(IXenConnection connection)
                 LoadTasksAsMeddlingActions(connection);
             }
 
+            // Allow Citrix Hypervisor Center connect to Havana, Stockholm and cloud released CH only

```

Comment:
I think this whole block should move to line 975. Regarding the if conditions, the first check is not needed because it is always true. The last check allows havana, but also ely and honolulu which we probably want to exclude (@MihaelaStoica could you please confirm), so let's try adding to it version checks from the Helpers class.

Reply:
Fixed.